### DISCUSSION TITLE
How to import functions from modules that contain React hooks import in RSC files?

### DISCUSSION THREAD

**Discussion by remorses** [vercel/next.js#49322] [App Router]
It's common to mix functions with React hooks in a single `utils.js` file, but if i try to import a function called x for example in a server component i will get the error 

```
Error: Attempted to call x() from the server but x is on the client. It's not possible to invoke a client function from the server, it can only be rendered as a Component or passed to props of a Client Component.
```

It should be possible for Next to detect that i am only importing a function that doesn't depend on React hooks, this could be done with dead code elimination

**Comment by remorses** [comment]
The same could be done with this error:

```
You're importing a component that needs useState. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.
```

Only throw this error after dead code elimination

**Comment by TunjiDev** [comment]
Come on, no responses really? facing the same issues too

**Comment by remorses** [comment]
Relevant discussion: https://github.com/apollographql/apollo-client/issues/10974#issuecomment-1594894341
